src/libcore/char/mod.rs

```diff
@@ -472,8 +482,26 @@ impl Iterator for CaseMappingIter {
             CaseMappingIter::Zero => None,
         }
     }
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match *self {
+            CaseMappingIter::Three(_, _, _) => {
+                (3, Some(3))
+            }
+            CaseMappingIter::Two(_, _) => {
+                (2, Some(2))
+            }
+            CaseMappingIter::One(_) => {
+                (1, Some(1))
+            }
+            CaseMappingIter::Zero => (0, Some(0))
```

Comment:
While I'm confident this is correct, I think it'd still be good to have a test to pin the behaviour; a bunch of things like `assert_eq!('a'.to_lowercase().len(), 1)`, perhaps.